### ISSUE TITLE
How to access current props from a selector?

### ISSUE THREAD

**Issue by rkeeler** [keajs/kea#19]
I'm trying to create a selector that takes an id that is passed to the component through props and uses that id to pull the object out of a kea reducer that has that id. This is a case where i want to use a selector but I can't figure out how to make the selector get recalculated whenever the prop changes. 

Here's an example to demonstrate what i'm trying to accomplish. There is a `books` reducer that has a dictionary of books based on their id.
```js
books = {
    0: { title: "My Book" },
    1: { title: "My Other Book" },
}
```

Now I have a component that gets passed a book id:
```js
<BookDetail bookId={0} />
<BookDetail bookId={1} />
```

I want to intercept the `bookId` that was passed to BookDetail and inject a `book` prop. However, the selector i have defined doesn't get called for each BookDetail component. It only gets called once because the input selector hasn't changed. The only input selector is the `books` reducer, which hasn't changed in this case. Is there a way to tell kea to reevaluate a selector when a prop changes?

```js
@kea({
  reducers: ({ actions }) => ({
    books: [null, PropTypes.object, {}],
  }),
  selectors: ({ selectors, props }) => ({
    book: [
      () => [selectors.books],
      books => books[props.bookId],
      PropTypes.object,
    ],
  }),
});
```

Any help you can give would be appreciated. I've liked kea so far but it feels like what i'm trying to do shouldn't be difficult because it's so easy to do with reselect.

**Comment by mariusandra** [comment]
Hey, this was indeed not possible. Most of my own code was written before the `@kea` syntax was available (there was only `@connect`), so passing `props` to selectors was not something I thought about. Thank you for bringing this to the forefront!

I checked how reselect does it and implemented something similar: I added `props` as the second input to all selectors. Thus since `v0.20.3` this will now work:

```js
  selectors: ({ selectors }) => ({
    book: [
      () => [selectors.books, (_, props) => props.bookId], // input
      (books, bookId) => books[bookId], // combiner
      PropTypes.object // output
    ]
  })
```

The `selectors` object is (*currently!*) built to create selectors that depend on other selectors... not just simple selectors that take the `state` + `props` and return an object. Thus the verbosity.

In fact, the "input" and "combiner" are passed directly to reselect's `createSelector`. Selectors created via `createSelector` only update when their input selectors are changed, thus it is currently necessary to create a selector that only outputs the right prop (`(_, props) => props.bookId`), and use that as one of the inputs.

The full code that I tried and got working [is this](https://gist.github.com/mariusandra/eecec12edaa2d051af33e23b1eaf261b).

As an aside, I recommend you keep your books themselves in a different logic store than the one that's connected to your BookDetail component: [something like this](https://gist.github.com/mariusandra/a76dbb1a4d3a7a117a56cfcf5d720ad3).
